app/scripts/controllers/logs.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+/* global moment: false */
+
+angular.module('openhimConsoleApp')
+  .controller('LogsCtrl', function ($scope, $location, Api) {
+    function formatLog(log) {
+      return new moment(log.timestamp).format('YYYY-MM-D HH:mm:ss.SSS') + ' - ' + log.level + ': [' + log.label + '] ' + log.message + '\n';
+    }
+
+    var lastFetch;
+    var locParams = $location.search();
+    $scope.params = angular.equals({}, locParams) ? { level: 'info' } : locParams;
+    $scope.logs = '';
+
+    if ($scope.params.from || $scope.params.until) {
+      $scope.autoupdate = false;
+      $scope.tailLogs = false;
+    } else {
+      $scope.autoupdate = true;
+      $scope.tailLogs = true;
+    }
+
+    // fetch initial logs
+    $scope.logs = '';
+    lastFetch = new Date();
+
+    var fromDate, untilDate;
+    if ($scope.params.from && $scope.params.from.length > 0) {
+      fromDate = moment($scope.params.from).format();
+    }
+    if ($scope.params.until && $scope.params.until.length > 0) {
+      untilDate = moment($scope.params.until).format();
+    }
+
+    var localParam = {
+      from: fromDate,
+      until: untilDate,
+      level: $scope.params.level
+    };
+    if (!$scope.params.until) {
+      localParam.until = lastFetch.toISOString();
+    }
+    Api.Logs.query(localParam, function (results) {
+      results.forEach(function (log) {
+        $scope.logs += formatLog(log);
+      });
+    });
+
+    function fetchMoreLogs() {
+      var now = new Date();
+      var localParam = {
+        from: new Date(lastFetch).toISOString(),
+        until: now.toISOString(),
+        level: $scope.params.level
+      };
+      lastFetch = now;
+      Api.Logs.query(localParam, function (results) {
+        results.forEach(function (log) {
+          $scope.logs += formatLog(log);
```

Comment:
I was thinking that if you leave the page open and let it keep auto appending it might make that string really big after a while - although to be fair I'm not sure if it'll actually be a problem in practice.